### PR TITLE
feat(`anvil`): `--cache-path`

### LINKED ISSUE #3512: feat(`anvil`): be able to specify a prefix path for temp disk storage
### Component

Anvil

### Describe the feature you would like

When anvil state grows, it persists it temporarily to disk. I'd like to be able to specify a prefix path for the directory that gets created by anvil for this state.

### Additional context

I am running Anvil in the cloud, and want to attach a persistent disk to the instance and mount it at the path anvil will use for storing state. Because it's a random directory each time, there's no way for me to programatically mount my disk at that location before anvil starts.

### LINKED ISSUE #8654: feat(anvil): Set anvil fork cache location (`--cache-path`)
### Component

Anvil

### Describe the feature you would like

Currently the cache location is fixed to `~/.foundry/anvil/rpc`. It would really simplify the workflow (for example for running fork tests on CI) if we could set custom directory for the cache location. This is related to #4989 however the issue was related to taking too much space.

Another use case would be to save cache on different drive without the need for symlinking. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #3512 + Closes #8654 

## Solution

Introduces the `--cache-path` flag to anvil letting users can configure where the old state is stored

### CODE REVIEW

File: crates/anvil/src/config.rs [hunk 1]
```diff
@@ -969,6 +972,13 @@ impl NodeConfig {
         self
     }
 
+    /// Sets the path where old states are cached
```
Comment (zerosnacks) [inline crates/anvil/src/config.rs:975]:
```suggestion
    /// Sets the path where states are cached
```

File: crates/anvil/src/config.rs [hunk 2]
```diff
@@ -189,6 +189,8 @@ pub struct NodeConfig {
     pub alphanet: bool,
     /// Do not print log messages.
     pub silent: bool,
+    /// The path where old states are cached.
+    pub cache_path: Option<PathBuf>,
```
Comment (zerosnacks) [inline crates/anvil/src/config.rs:193]:
```suggestion
    /// The path where states are cached.
    pub cache_path: Option<PathBuf>,
```

**Review by zerosnacks — COMMENTED** [review]
Implementation LGTM, would you mind adding a test case for this?

File: crates/anvil/tests/it/anvil.rs [hunk 3]
```diff
@@ -118,3 +121,20 @@ async fn test_cancun_fields() {
...
+async fn test_cache_path() {
+    run_with_tempdir("custom-anvil-cache", |tmp_dir| async move {
+        let cache_path = tmp_dir.join("cache");
+        let (_api, _handle) = spawn(
+            NodeConfig::test()
+                .with_cache_path(Some(cache_path.clone()))
+                .with_max_persisted_states(Some(10_usize))
+                .with_blocktime(Some(Duration::from_millis(1))),
+        )
+        .await;
```
Comment (grandizzy) [inline crates/anvil/tests/it/anvil.rs:135]: is there a way to assert file was actually written in custom temp dir? I tried to manually test it but probably I am missing something as I wasn't able to get the json file dumped

**Review by grandizzy — COMMENTED** [review]
lgtm! one minor question re testing the new setting

File: crates/anvil/tests/it/anvil.rs (hunk 3 above)
Comment (yash-atreya) [reply crates/anvil/tests/it/anvil.rs:136]: Yeah so testing this is a bit hard. Wrote a better test here. [363010f](https://github.com/foundry-rs/foundry/pull/9343/commits/363010fe48c4e93085e1f5537626404d71b1e5e5)

Comment (grandizzy) [reply crates/anvil/tests/it/anvil.rs:136]: thank you! yeah, managed to check file is generated with this test. I think that's good to have, drawback is that will slow down the CI (and potential flakiness), @zerosnacks wdyt?

If we are to keep it let's solve warnings here
https://github.com/foundry-rs/foundry/actions/runs/11933626813/job/33261072474#step:12:66

and maybe add one more assert to make sure the cache is actually populated
```rust
assert!(cache_path.read_dir().unwrap().next().is_some());
```

**Review by grandizzy — APPROVED** [review]
Lgtm!

**Comment by grandizzy** [comment]
looks like the test is flaky, CI fails here https://github.com/foundry-rs/foundry/actions/runs/11948242152/job/33309461327#step:12:151 I propose to remove it, @zerosnacks @yash-atreya  wdyt?

**Pull request merged**